File: Algorithms/Recursive Algorithms/C/towerOfHanoi.c
#include<stdio.h>
int cnt=0;

void move(int,char,char,char);

int main(){
	int disk;
	printf("Input num of disks : ");
	scanf("%d",&disk);
	move(disk,'A','B','C');
	return 0;
}

void move(int n, char a, char b, char c)
{
	if (n == 1)
	{
		++cnt;
		printf("\n%5d: Move disk 1 from %c to %c", cnt, a, c);
		return;
	}
	else
	{
		move(n-1, a, c, b);
		++cnt;
		printf("\n%5d: Move disk %d from %c to %c", cnt, n, a, c);
		move(n-1, b, a, c);
		return;
	}
}

File: Algorithms/Recursive Algorithms/Go/tower_of_hanoi.go
package main

import (
	"fmt"
	"strconv"
)

func TOH_moves(num int, start, inter, end string) {
	if num == 1 {
		fmt.Println("Move disk 1 from " + start + " to " + end)
		return
	}
	TOH_moves(num-1, start, end, inter)
	fmt.Println("Move disk " + strconv.Itoa(num) + " from " + start + " to " + end) //strconv.Itoa() used for type casting int to string
	TOH_moves(num-1, inter, end, start)

}

func main() {

	var num int

	fmt.Print("Enter the  number of disk : ")
	fmt.Scanln(&num)
	TOH_moves(num, "A", "B", "C")
}
